### DISCUSSION TITLE
follow upstream redirects directive

### DISCUSSION THREAD

**Discussion by ahmadnassri** [Kong/kong#7219] [Ideas and feature requests]
if the upstream responds with a redirect, typically it will send an absolute path that will not match the one Kong clients are calling _(both request host and request path)_.

a `follow_redirects` directive on the API would instruct nginx to retry the same request at the new location **before** sending back to the client, thus eliminating the need for the client to know about the redirect.

**Comment by thibaultcha** [comment]
Should probably use [`proxy_redirect`](http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_redirect).

**Comment by catchdave** [comment]
Proxy_redirect would fix the issue nicely :)

**Comment by 3dbrows** [comment]
+1, this would be particularly useful when the redirect from the upstream host is unreachable from the end user's machine, but is reachable by Kong (e.g. Kong is exposing an internal service).

**Comment by anamarialazar** [comment]
+1

**Comment by JnMik** [comment]
+1 Need this too

**Comment by lnryan** [comment]
+1 also need this

**Comment by kl4n4** [comment]
+1 same here

**Comment by john-deng** [comment]
+1 need this too

**Comment by jakebolam** [comment]
+1 looking for this

**Comment by MingStar** [comment]
+1 this would be a much-needed option in kong itself (instead of fiddling with the underlying nginx configs)
